File: YuGiOh2/Cards/C22359980.cs
﻿using System.Linq;
using YuGiOh2.Base;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 银幕之镜壁
    /// </summary>
    public class C22359980
    {
        public static int Type { get; } = (int)AffectMomentType.WhenAttacked;

        public static void ProcessEffect(string targetID, Player player)
        {
            player.Enemy.Field.MonsterFields
                .Where(c => c != null && !c.Status.FaceDown)
                .ToList()
                .ForEach(c => c.ATK >>= 1);
        }
    }
}

File: YuGiOh2/Scripts/C22359980.lua
Type = 2;
--银幕之镜壁
function processEffect(targetID, player)
    for index, monster in pairs(player.Enemy.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.FaceDown then
            monster.ATK = monster.ATK / 2;
        end
    end
end